src/sentry/auth/idpmigration.py

```diff
@@ -14,60 +15,79 @@
 _TTL = timedelta(minutes=10)
 
 
-def send_confirm_email(user: User, email: str, verification_key: str) -> None:
-    context = {
-        "user": user,
-        "url": absolute_uri(
-            reverse(
-                "sentry-idp-email-verification",
-                args=[verification_key],
-            )
-        ),
-        "confirm_email": email,
-        "verification_key": verification_key,
-    }
-    msg = MessageBuilder(
-        subject="{}Confirm Email".format(options.get("mail.subject-prefix")),
-        template="sentry/emails/idp_verification_email.txt",
-        html_template="sentry/emails/idp_verification_email.html",
-        type="user.confirm_email",
-        context=context,
-    )
-    msg.send_async([email])
-
-
 def send_one_time_account_confirm_link(
-    user: User, org: Organization, email: str, identity_id: str
+    user: User,
+    org: Organization,
+    provider_name: str,
+    email: str,
+    identity_id: str,
 ) -> str:
-    """Store and email a verification key for IdP migration.
-
-    Create a one-time verification key for a user whose SSO identity
-    has been deleted, presumably because the parent organization has
-    switched identity providers. Store the key in Redis and send it
-    in an email to the associated address.
-
-    :param user: the user profile to link
-    :param org: the organization whose SSO provider is being used
-    :param email: the email address associated with the SSO identity
-    :param identity_id: the SSO identity id
-    """
-    cluster = redis.clusters.get("default").get_local_client_for_key(_REDIS_KEY)
-    member_id = OrganizationMember.objects.get(organization=org, user=user).id
-
-    verification_code = get_random_string(32, string.ascii_letters + string.digits)
-    verification_key = f"auth:one-time-key:{verification_code}"
-    verification_value = {
-        "user_id": user.id,
-        "email": email,
-        "member_id": member_id,
-        "identity_id": identity_id,
-    }
-    cluster.hmset(verification_key, verification_value)
-    cluster.expire(verification_key, int(_TTL.total_seconds()))
-
-    send_confirm_email(user, email, verification_code)
-
-    return verification_code
+    return AccountConfirmLink(user, org, provider_name, email, identity_id).send_one_time_link()
+
+
+@dataclass
+class AccountConfirmLink:
+    user: User
+    organization: Organization
+    provider_name: str
+    email: str
+    identity_id: str
```

Comment:
I hope this refactoring doesn't seem gratuitous. Otherwise, it would have been necessary to duplicate both the org and provider name between the parameter lists for the two functions, which tripped my personal "time to gather everything into a class" threshold.

The "hide whitespace changes" setting should cut down on the diff noise.